universal-application-tool-0.0.1/app/views/admin/questions/QuestionsListView.java

```diff
@@ -0,0 +1,97 @@
+package views.admin.questions;
+
+import static j2html.TagCreator.a;
+import static j2html.TagCreator.body;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.each;
+import static j2html.TagCreator.table;
+import static j2html.TagCreator.tbody;
+import static j2html.TagCreator.th;
+import static j2html.TagCreator.thead;
+import static j2html.TagCreator.tr;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import j2html.tags.Tag;
+import play.twirl.api.Content;
+import services.question.QuestionDefinition;
+import views.BaseHtmlLayout;
+import views.BaseHtmlView;
+
+public final class QuestionsListView extends BaseHtmlView {
+  private final BaseHtmlLayout layout;
+  private final ImmutableList<QuestionRow> tableCells =
+      ImmutableList.of(
+          QuestionRow.ID,
+          QuestionRow.VERSION,
+          QuestionRow.PATH,
+          QuestionRow.NAME,
+          QuestionRow.DESCRIPTION,
+          QuestionRow.QUESTION_TEXT,
+          QuestionRow.QUESTION_HELP_TEXT,
+          QuestionRow.QUESTION_TYPE,
+          QuestionRow.ACTIONS);
+
+  @Inject
+  public QuestionsListView(BaseHtmlLayout layout) {
+    this.layout = layout;
+  }
+
+  /** Renders a page with either a table or a list view of all questions. */
+  public Content render(ImmutableList<QuestionDefinition> questions, String renderAs) {
+    return layout.htmlContent(
+        body()
+            .with(renderHeader("All Questions"))
+            .with(renderAllQuestions(questions, renderAs))
+            .with(renderSummary(questions))
+            .with(renderAddQuestionLink()));
+  }
+
+  private Tag renderAddQuestionLink() {
+    return a("Create a new question").withHref("/admin/questions/new");
```

Comment:
Suggestion (blocking): use reverse router for path

Reply:
Done.